src/lib.rs

```diff
@@ -950,6 +950,9 @@ impl<
             api: c_api.clone(),
         };
         let _ = da_member.run_view().await;
+        // TODO (da) Replica task isn't added since the proposal it listens to is the commitment
+        // proposal but `I::Proposal` here is the DA proposal. We need to support the two proposal
+        // types for sequencing consensus in the refactored node implementation.
```

Comment:
Can we explicitly pass the Proposal type as a generic parameter to `create_or_obtain_chan_from_read` and `create_or_obtain_chan_from_write` and then use that to create one recv channel for each proposal type?

Reply:
Hmm also the `channel_map` is typed with a signle proposal type so yeah it's not very simple to do